Utils/tests/Expect/iterableElementRuntimeIsNot_A01_t01.dart

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion static void iterableElementsRuntimeIsNot<T>(dynamic it)
+/// Failure if `it` is an instance of `Iterable<T>` and has an element type of
+/// which is not exactly the `T`
+///
+/// @description Checks no exception is thrown when `it` is an `Iterable<T>` and
+/// all elements of it have a type `T` (not a subtype)
+/// @author sgrekhov22@gmail.com
+
+import "dart:async";
+import "../../../Utils/expect.dart";
+
+main() {
```

Comment:
If `iterableElementsRuntimeIs` and `...Not` are updated then this test might just remain unchanged. However, I suspect that there won't be a `...Not`, because that one should be covered by the other one.

Also, it should probably be called `runtimeTypeImplementsIterableOf<X>()` and invoked as `<num>[].runtimeTypeImplementsIterableOf<T>()`, and it would succeed when `T` is `num` and fail when `T` is `Object` or `int`.